sxt/base/field/subtract_p.h

```diff
@@ -27,30 +27,29 @@
 
 #include "sxt/base/field/arithmetic_utility.h"
 #include "sxt/base/macro/cuda_callable.h"
-#include "sxt/field12/base/constants.h"
 
-namespace sxt::f12b {
+namespace sxt::basf {
 //--------------------------------------------------------------------------------------------------
 // subtract_p
 //--------------------------------------------------------------------------------------------------
-CUDA_CALLABLE inline void subtract_p(uint64_t ret[6], const uint64_t a[6]) noexcept {
+/*
+ Compute ret = a - p, where p is the modulus.
+ */
+template <const size_t limbs>
+CUDA_CALLABLE inline void subtract_p(uint64_t* ret, const uint64_t* const a,
+                                     const uint64_t* const p) noexcept {
```

Comment:
Please use a template parameter like
```
template <size_t NumLimbs>
```
Using const on value template parameters is unconventional and doesn't do anything as you can't modify them anyways.

Please change arguments to
```
const uint64_t* const a -> const uint64_t* a
const uint64_t const p -> const uint64_t* p
```
See note on function parameters
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#con1-by-default-make-objects-immutable

Reply:
Updated.